hackage-security/src/Hackage/Security/Util/IO.hs

```diff
@@ -21,13 +21,13 @@ import Hackage.Security.Util.Path
 getFileSize :: (Num a, FsRoot root) => Path root -> IO a
 getFileSize fp = fromInteger <$> withFile fp ReadMode hFileSize
 
-handleDoesNotExist :: IO a -> IO (Maybe a)
-handleDoesNotExist act =
-   handle aux (Just <$> act)
+handleDoesNotExist :: a -> IO a -> IO a
```

Comment:
this change is actually a breaking change (since it's an exposed function) and would demand a major ver bump!